# Copyright 2022 Shadow Robot Company Ltd.
#
# This program is free software: you can redistribute it and/or modify it
# under the terms of the GNU General Public License as published by the Free
# Software Foundation version 2 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
# more details.
#
# You should have received a copy of the GNU General Public License along
# with this program. If not, see <http://www.gnu.org/licenses/>.

sr_calibrations: [
  ["FFJ1", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #FFJ1
  ["FFJ2", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #FFJ2
  ["FFJ3", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #FFJ3
  ["FFJ4", [ [0.0, 0.0],
             [1.0,  1.0] ]],   #FFJ4

  ["MFJ1", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #MFJ1
  ["MFJ2", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #MFJ2
  ["MFJ3", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #MFJ3
  ["MFJ4", [ [0.0, 0.0],
             [1.0,  1.0] ]],   #MFJ4

  ["RFJ1", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #RFJ1
  ["RFJ2", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #RFJ2
  ["RFJ3", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #RFJ3
  ["RFJ4", [ [0.0, 0.0],
             [1.0,  1.0] ]],   #RFJ4

  ["LFJ1", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #LFJ1
  ["LFJ2", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #LFJ2
  ["LFJ3", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #LFJ3
  ["LFJ4", [ [0.0, 0.0],
             [1.0,  1.0] ]],   #LFJ4
  ["LFJ5", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #LFJ5


  ["THJ1", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #THJ1
  ["THJ2", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #THJ2
  ["THJ3", [ [0.0,  0.0],
             [1.0, 1.0] ]],   #THJ3
  ["THJ4", [ [0.0,  0.0],
             [1.0,  1.0] ]],   #THJ4
  ["THJ5", [ [0.0,  0.0],
             [1.0,  1.0] ]],   #THJ5

  ["WRJ1", [ [0.0, 0.],
             [1.0,  0.] ]],    #WRJ1
  ["WRJ2", [ [0.0, 0.],
             [1.0,  0.] ]]     #WRJ2
]

sr_calibrations_coupled: [
  [["THJ1", "THJ2"], [[[2738.0, 2301.0], 0.0, 40.0],
                      [[2693.0, 2154.0], 0.0, 20.0],
                      [[2680.0, 1978.0], 0.0, 0.0],
                      [[2677.0, 1840.0], 0.0, -20.0],
                      [[2664.0, 1707.0], 0.0, -40.0],
                      [[2334.0, 2287.0], 22.5, 40.0],
                      [[2242.0, 2095.0], 22.5, 20.0],
                      [[2230.0, 1953.0], 22.5, 0.0],
                      [[2223.0, 1807.0], 22.5, -20.0],
                      [[2206.0, 1685.0], 22.5, -40.0],
                      [[1839.0, 2243.0], 45.0, 40.0],
                      [[1772.0, 2112.0], 45.0, 20.0],
                      [[1764.0, 1928.0], 45.0, 0.0],
                      [[1755.0, 1762.0], 45.0, -20.0],
                      [[1683.0, 1650.0], 45.0, -40.0],
                      [[1387.0, 2219.0], 67.5, 40.0],
                      [[1375.0, 2056.0], 67.5, 20.0],
                      [[1370.0, 1884.0], 67.5, 0.0],
                      [[1337.0, 1741.0], 67.5, -20.0],
                      [[1329.0, 1630.0], 67.5, -40.0],
                      [[1141.0, 2206.0], 90.0, 40.0],
                      [[1132.0, 2055.0], 90.0, 20.0],
                      [[1114.0, 1877.0], 90.0, 0.0],
                      [[1103.0, 1730.0], 90.0, -20.0],
                      [[1092.0, 1615.0], 90.0, -40.0]]]  #THJ1 & THJ2
]
